src/service_info.rs

```diff
@@ -642,7 +642,11 @@ fn decode_txt(txt: &[u8]) -> Vec<TxtProperty> {
         }
         offset += 1; // move over the length byte
 
-        let kv_bytes = &txt[offset..offset + length];
+        let offset_end = offset + length;
+        if offset_end > txt.len() {
+            break; // Would be out of range, skipping this property.
```

Comment:
Make sense! Could you also add a log of `offset_end` and `txt.len()` when this happens?  I would say `error!()` level.  And if `error!()` doesn't work for you, `debug!()` level is also fine. 

Can you also add a unit test in this file inside `mod tests` for this negative case?  A simple test input would be fine.

Reply:
Gotcha, I've give given it a go!